### ISSUE TITLE
bound_projects property not set for gcp_auth_backend_role

### ISSUE THREAD

**Issue by msotoperdomo** [terraform-providers/terraform-provider-vault#394]
Hi there,

### Terraform Version
```
Terraform v0.11.13
+ provider.google v2.3.0
+ provider.vault v1.7.0

Vault server version 1.1.0.
```

### Affected Resource(s)
Please list the resources as a list, for example:
- vault_gcp_auth_backend_role

### Terraform Configuration Files
```hcl
resource "vault_gcp_auth_backend" "backend" {
  path        = "gcp/example/module-test"
  credentials = "${base64decode(google_service_account_key.test_module_gcp_vault_backend_sa_key.private_key)}"
  description = "This is the backend for example/module-test"
}

resource "vault_gcp_auth_backend_role" "gcp" {
  role                   = "module-test"
  type                   = "iam"
  backend                = "${vault_gcp_auth_backend.backend.path}"
  bound_projects         = ["example"]
  bound_service_accounts = ["${google_service_account.test_module_gcp_vault_backend_sa.email}"]
  policies               = ["default"]
  ttl                    = "600"
  max_ttl                = "2800"
  period                 = "3600"
  max_jwt_exp            = "2800"
}

resource "google_service_account" "test_module_gcp_vault_backend_sa" {
  account_id = "test-mdule-gcp-sa"
  project    = "example"
}

resource "google_service_account_key" "test_module_gcp_vault_backend_sa_key" {
  service_account_id = "${google_service_account.test_module_gcp_vault_backend_sa.name}"
}
```
### Expected Behavior
Once applied the `bound_projects` property is created in the vault server.

### Actual Behavior
The vault gcp role is created in vault but `bound_projects` is missing. When trying to apply again to "fix" it, terraform does not find any difference to be applied.
```
vault read  auth/gcp/example/module-test/role/module-test
Key                       Value
---                       -----
add_group_aliases         false
allow_gce_inference       true
bound_service_accounts    [test-mdule-gcp-sa@example.iam.gserviceaccount.com]
max_jwt_exp               2800
max_ttl                   46m40s
period                    1h
policies                  [default]
ttl                       10m
type                      iam
```

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform init`
1. `terraform apply`

### References
- [PR-278](https://github.com/terraform-providers/terraform-provider-vault/pull/278)

### Other scenarios
1. When using project_id instead of bound_projects the behaviour is the same.
2. Tested also with v1.6.0 same behaviour.

**Comment by nathanwilk7** [comment]
I think I may have just seen an issue with the same root cause. With `provider.vault` version `1.7.0` I was unable to create a `vault_gcp_auth_backend_role` with `terraform apply`. However, when I downgraded my `provider.vault` version to `1.5.0`, I was able to successfully create a `vault_gcp_auth_backend_role`.

**Comment by hmakkapati** [comment]
Seeing the same issue with 1.6.0+
Thinking this is the issue https://github.com/terraform-providers/terraform-provider-vault/pull/308/files#diff-0ea2251b55e7d42328544560a44846efL143

Going to attempt a quick fix.

**Comment by hmakkapati** [comment]
Fix is available in `1.9.0` https://github.com/terraform-providers/terraform-provider-vault/commit/271ee83e2fdca28e5f66f6e209fe975eb5fb3ef6

**Comment by msotoperdomo** [comment]
Confirmed is working with vault provider version 1.9.0 and terraform 0.11.13 👍
